pkg/storage/util/util.go

```diff
@@ -50,18 +53,47 @@ func UpdateCondition(cr *imageregistryv1.Config, conditionType string, status op
 	cr.Status.Conditions = conditions
 }
 
-func GetClusterVersionConfig(kubeconfig *rest.Config) (*configv1.ClusterVersion, error) {
-	client, err := configv1client.NewForConfig(kubeconfig)
+// GetInfrastructure gets information about the cloud platform that the cluster is
+// installed on including the Type, Region, and other platform specific information.
+// Currently the install config is used as a backup to be compatible with upgrades
+// from 4.1 -> 4.2 when platformStatus did not exist, but should be able to be removed
+// in the future.
+func GetInfrastructure(listers *regopclient.Listers) (*configv1.Infrastructure, error) {
+	infra, err := listers.Infrastructures.Get("cluster")
 	if err != nil {
```

Comment:
1) Shouldn't we check for not found here?
2) In a 4.1 -> 4.2 scenario, will we ever get to this point? Will the lister cache even populate because we need the CVO to create the Infrastructures CRD?

Reply:
I believe that infrastructures was already there in 4.1, it's not new to 4.2, what is new is PlatformStatus, which was Platform in 4.1, which we are setting up here if it's not found.  Does that help?